Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -111,6 +114,53 @@ void TRDGlobalTracking::updateTimeDependentParams()
   mTracker->SetTPCVdrift(mTPCVdrift);
 }
 
+void TRDGlobalTracking::fillMCTruthInfo(const TrackTRD& trk, const o2::MCCompLabel& lblSeed, std::vector<o2::MCCompLabel>& lblContainer, const o2::dataformats::MCTruthContainer<o2::MCCompLabel>* trkltLabels) const
+{
+  // Check MC labels of the TRD tracklets attached to the track seed.
+  // Set TRD track label to the most frequent tracklet label.
+  // Fake flag is set if either not all TRD tracklets have most frequent label
+  // or if the seeding label is different from the most frequent TRD label.
+  // In case multiple tracklet labels occur most often we choose the one which matches the label of the seed, or,
+  // if that is not the case one of the most frequent labels is chosen arbitrarily
+  LOG(DEBUG) << "Checking seed with label: " << lblSeed;
+  std::unordered_map<o2::MCCompLabel, unsigned int> labelCounter;
+  int nTracklets = 0;
+  unsigned int maxOccurences = 0;
+  for (int iLy = 0; iLy < constants::NLAYER; ++iLy) {
+    auto trkltIndex = trk.getTrackletIndex(iLy);
+    if (trkltIndex == -1) {
+      // no tracklet in this layer
+      continue;
+    }
+    const auto& lblsTrklt = trkltLabels->getLabels(trkltIndex);
+    for (const auto lblTrklt : lblsTrklt) {
+      ++labelCounter[lblTrklt];
+      if (labelCounter[lblTrklt] > maxOccurences) {
+        maxOccurences = labelCounter[lblTrklt];
+      }
+    }
+  }
+  o2::MCCompLabel mostFrequentLabel;
+  for (const auto& [lbl, count] : labelCounter) {
+    LOG(DEBUG) << "Label " << lbl << " occured " << count << " times.";
+    if (count == maxOccurences) {
+      if (lblSeed.compare(lbl) >= 0) {
+        // most frequent label matches seed label
+        mostFrequentLabel = lbl;
+        mostFrequentLabel.setFakeFlag(maxOccurences != trk.getNtracklets());
```

Comment:
The convention (unfortunately, undocumented yet) for the matches label is that they have the MC track ID of the source track (effectively, that of the TPC, since the same is applied to ITS-TPC tracks) and isFake is set only if the mostFrequentLabel (even if by itself it would have a fake flag set)  does not match to that ID.
It is somewhat unfortunate that these labels have a meaning of both the track and match label. For the ITS and ITSAB (which is just a collection of the cluster references, we have their proper labels (with the fake flag set if mostFrequentLabel is not seen on every contributing layer) and there is a separate label for the TrackTPCITS. 
Since you don't have a separate TRD-info only track object, perhaps you could add simply a 2nd vector of labels which would bear the label you create now, while the label associated with the match should be `lblSeed.setFakeFlag(lblSeed == mostFrequentLabel)`

Reply:
Ok I created two label containers now:
1. `lblMatch` is simply the label of the seeding track and the fake flag is set if `lblSeed != mostFrequentLabel`.
2. `lblTrd` is the most frequent TRD label. Should I set the fake flag here only if the most frequent label does not appear in every layer where a tracklet is attached? I think this would make more sense than also requesting that the label matches the seed's label, since it is supposed to be the label of the TRD track.

And I am also wondering how to name the message with the TRD track labels. I need two different descriptions depending on the seed, so I thought `MCLB_ITSTPC_TRD` and `MCLB_TPC_TRD` would be ok. Or should I use `MCLB_TRD` with different subspecs?

Reply:
Hm, `MCLB_TRD` with different subspecs is not good, since the subspec is used to indicate strict matching mode. So I will go with `MCLB_ITSTPC_TRD`

Reply:
Hi, yes, for the TRD label proper you should ignore the trackID of the seeding track, just the dominant contributor and its seFake(true) if it is not represented on every layer.
MCLB_ITSTPC_TRD and MCLB_TPC_TRD should be ok, indeed, better to not touch the subspec.